docs/docs/validating-plugin-options.md

```diff
@@ -0,0 +1,230 @@
+---
+title: Validating plugin options
+---
+
+To help users [configure plugins](/docs/configuring-usage-with-plugin-options/) correctly, a plugin can optionally define a schema to enforce a type for each option. When users use a plugin, Gatsby will validate that the provided options match the schema.
+
+## How to define an options schema
+
+To define their options schema, plugins export a [`pluginOptionsSchema`](/docs/node-apis/#pluginOptionsSchema) from their `gatsby-node.js`. This function gets passed an instance of [Joi](https://joi.dev) to define the schema with.
+
+For example, consider the following plugin called `gatsby-plugin-console-log`:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: `gatsby-plugin-console-log`,
+      options: {
+        optionA: true,
+        message: "Hello world"
+        optionB: false, // Optional.
+      },
+    },
+  ],
+}
+```
+
+`gatsby-plugin-console-log` can ensure users have to pass a boolean to `optionA` and a string to `message`, as well as optionally pass a boolean to `optionB`, with this `pluginOptionsSchema`:
+
+```javascript:title=plugins/gatsby-plugin-console-log/gatsby-node.js
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    optionA: Joi.boolean().required().description(`Enables optionA.`),
+    message: Joi.string()
+      .required()
+      .description(`The message logged to the console.`),
+    optionB: Joi.boolean().description(`Enables optionB.`),
+  })
+}
+```
+
+If users pass options that do not match the schema, the validation will show an error when they run `gatsby develop` and prompt them to fix their configuration.
+
+## Unit testing an options schema
+
+To verify that a `pluginOptionsSchema` behaves as expected, unit test it with different configurations using the [`gatsby-plugin-utils` package](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-utils#testpluginoptionsschema).
+
+1. Add the `gatsby-plugin-utils` package to your site:
+
+   ```shell
+   yarn add gatsby-plugin-utils
+   ```
+
+   or
+
+   ```shell
+   npm install gatsby-plugin-utils
+   ```
```

Comment:
````suggestion
   ```shell
   npm install gatsby-plugin-utils
   ```
````

We're only ever using `npm` in our docs :)